lib/nexus/nexus_plot.cpp

```diff
@@ -85,30 +101,87 @@ hinfo headerinfo( std::istream& stream ) {
     }
 
     stream.seekg(8, std::ios::cur);
-    stream.read((char*) h.vars_in_class.data(), h.num_classes * 4);
-    std::transform( h.vars_in_class.begin(),
-                    h.vars_in_class.end(),
-                    h.vars_in_class.begin(),
-                    ntoh );
-    if (std::any_of( h.vars_in_class.begin(), h.vars_in_class.end(), negative))
+    std::vector< int32_t > vars_in_class( h.num_classes );
+    stream.read((char*) vars_in_class.data(), h.num_classes * 4);
+    std::transform( vars_in_class.begin(),
+                    vars_in_class.end(),
+                    vars_in_class.begin(),
+                    ntohl );
+    if (std::any_of( vars_in_class.begin(), vars_in_class.end(), negative))
         throw nex::bad_header("Negative value, corrupted file");
+    for (int i = 0; i < h.num_classes; i++)
+        h.vars_in_class[h.class_names[i]] = vars_in_class[i];
 
     stream.seekg(8, std::ios::cur);
     for (int i = 0; i < h.num_classes; ++i) {
         stream.seekg(4, std::ios::cur);
-        std::vector< char > var_names( h.vars_in_class[i] * 4, 0 );
-        stream.read( var_names.data(), h.vars_in_class[i] * 4 );
-        for (int k = 0; k < h.vars_in_class[i]; ++k)
-            h.var_names[i].push_back( std::string( var_names.data() + k*4 ,4 ));
+        std::vector<char> var_names( h.vars_in_class[h.class_names[i]] * 4, 0 );
+        stream.read( var_names.data(), h.vars_in_class[h.class_names[i]] * 4 );
+        for (int k = 0; k < h.vars_in_class[h.class_names[i]]; ++k)
+            h.var_names[h.class_names[i]]
+                .push_back(std::string( var_names.data() + k*4 ,4 ));
         stream.seekg(8, std::ios::cur);
     }
-    stream.seekg(4, std::ios::cur);
 
     return h;
 }
 
+std::map< int, nex::NexusPlot::timestep_data > nexus_classes( hinfo h,
+        std::istream& stream ) {
+    std::map< int, nex::NexusPlot::timestep_data > data;
+
+    std::string class_name = read_str<8>(stream);
+    while (class_name.compare("STOP    ")) {
+        //Skipped
+        stream.seekg(8, std::ios::cur);
+
+        std::array< int32_t, 5 > buf {};
+        stream.read( (char*)buf.data(), buf.max_size() * 4 );
+        if ( !stream.good() ) throw nex::unexpected_eof("");
+        std::transform( buf.begin(), buf.end(), buf.begin(), ntohl );
+
+        int timestep  = (int) int32_t_bits_to_float(buf[0]);
+        int num_items = (int) int32_t_bits_to_float(buf[2]);
```

Comment:
not really an issue, but I think the C-style type casting is kind of frowned upon...